trace/kafka/kafka.go

```diff
@@ -7,14 +7,21 @@ import (
 	"github.com/beatlabs/patron/encoding/json"
 	"github.com/beatlabs/patron/errors"
 	"github.com/beatlabs/patron/trace"
-	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go"
 	"github.com/opentracing/opentracing-go/ext"
 )
 
 // Message abstraction of a Kafka message.
 type Message struct {
 	topic string
 	body  []byte
+	key   *string
+}
+
+// SetKey sets the key of the message
+func (m *Message) SetKey(k *string) *Message {
```

Comment:
I propose to introduce 2 new constructors `NewMessageWithKey` and `NewJSONMessageWithKey`.
The messages should be `immutable` in a way so a constructor is a better approach,

Reply:
Yes indeed, i have changed the implementation to multiple constructors.

Reply:
@slysterous you can remove the setter since you have added the constructors

Reply:
@mantzas silly me, it's ok now.